src/Hl7.Fhir.Core/Rest/BaseFhirClient.cs

```diff
@@ -0,0 +1,1025 @@
+
+            return GetAsync(new Uri(url, UriKind.RelativeOrAbsolute));
+        }
+
+        #endregion
+        
+        private ResourceIdentity verifyResourceIdentity(Uri location, bool needId, bool needVid)
+        {
+            var result = new ResourceIdentity(location);
+
+            if (result.ResourceType == null) throw Error.Argument(nameof(location), "Must be a FHIR REST url containing the resource type in its path");
+            if (needId && result.Id == null) throw Error.Argument(nameof(location), "Must be a FHIR REST url containing the logical id in its path");
+            if (needVid && !result.HasVersion) throw Error.Argument(nameof(location), "Must be a FHIR REST url containing the version id in its path");
+
+            return result;
+        }
+
+
+        // TODO: Depending on type of response, update identity & always update lastupdated?
+
+        private void updateIdentity(Resource resource, ResourceIdentity identity)
+        {
+            if (resource.Meta == null) resource.Meta = new Meta();
+
+            if (resource.Id == null)
+            {
+                resource.Id = identity.Id;
+                resource.VersionId = identity.VersionId;
+            }
+        }
+
+
+        private void setResourceBase(Resource resource, string baseUri)
+        {
+            resource.ResourceBase = new Uri(baseUri);
+
+            if (resource is Bundle)
+            {
+                var bundle = resource as Bundle;
+                foreach (var entry in bundle.Entry.Where(e => e.Resource != null))
+                    entry.Resource.ResourceBase = new Uri(baseUri, UriKind.RelativeOrAbsolute);
+            }
+        }
+
+
+        // Original
+        private TResource execute<TResource>(Bundle tx, HttpStatusCode expect) where TResource : Model.Resource
+        {
+            return executeAsync<TResource>(tx, new[] { expect }).WaitResult();
+        }
+        public Task<TResource> executeAsync<TResource>(Model.Bundle tx, HttpStatusCode expect) where TResource : Model.Resource
+        {
+            return executeAsync<TResource>(tx, new[] { expect });
+        }
+        // Original
+        private TResource execute<TResource>(Bundle tx, IEnumerable<HttpStatusCode> expect) where TResource : Resource
+        {
+            return executeAsync<TResource>(tx, expect).WaitResult();
+        }
+
+        private async Task<TResource> executeAsync<TResource>(Bundle tx, IEnumerable<HttpStatusCode> expect) where TResource : Resource
+        {
+            verifyServerVersion();
+
+            var request = tx.Entry[0];
+            // tx (-> ITyped)? -> entryRequest 
+            // entry -> ITyped -> tx
+            var entryRequest = request.ToEntryRequest(Settings);
+
+            var entryResponse = (await Requester.ExecuteAsync(entryRequest).ConfigureAwait(false)).ToTypedEntryResponse(Settings.ParserSettings, new PocoStructureDefinitionSummaryProvider());
```

Comment:
the creation of `PocoStructureDefinitionSummaryProvider` should better be in the constructor of this class, or should it be a parameter of the constructor?

Reply:
This will need to be an explicit dependency - so on the constructor.

Reply:
And maybe have a default for it, which is (for now) just this `new PocoStructureDefinitionSummaryProvider`.

Reply:
done, put it on the constructor and added an optional parameter on the constructor as well.